Sources/SwiftGRPCNIO/HTTP1ToRawGRPCServerCodec.swift

```diff
@@ -113,25 +155,82 @@ extension HTTP1ToRawGRPCServerCodec: ChannelInboundHandler {
 extension HTTP1ToRawGRPCServerCodec: ChannelOutboundHandler {
   public typealias OutboundIn = RawGRPCServerResponsePart
   public typealias OutboundOut = HTTPServerResponsePart
-  
+
   public func write(ctx: ChannelHandlerContext, data: NIOAny, promise: EventLoopPromise<Void>?) {
     let responsePart = self.unwrapOutboundIn(data)
     switch responsePart {
-    case .headers(let headers):
-      //! FIXME: Should return a different version if we want to support pPRC.
-      ctx.write(self.wrapOutboundOut(.head(HTTPResponseHead(version: .init(major: 2, minor: 0), status: .ok, headers: headers))), promise: promise)
+    case .headers:
+      var headers = HTTPHeaders()
+      let version: HTTPVersion
+      if let contentType = contentType {
+        headers.add(name: "content-type", value: contentType.rawValue)
+        switch contentType {
+        case .text, .web:
+          version = .init(major: 1, minor: 1)
+        case .binary:
+          version = .init(major: 2, minor: 0)
+        }
+      } else {
+        version = .init(major: 2, minor: 0)
```

Comment:
How about setting the content type to `.binary` if no header is present, to avoid logic like this?

Reply:
Done.